### DISCUSSION TITLE
Help starting mux session

### DISCUSSION THREAD

**Discussion by linsomniac** [wezterm/wezterm#2031] [Q&A] [answered]
I set up a mux session last week and it has been working great, but today I rebooted the remote end and now I'm having problems.

I'm running 20220408-101518-b908e2dd and have an ssh domain set up, I can "wezterm ssh nlaptop".  But if I "wezterm connect nlaptop", it says "20:25:19.265  ERROR  wezterm_client::client  > ssh stderr: 20:25:19.412  ERROR  wezterm > connecting to /run/user/1000/wezterm/sock: No such file or directory (os error 2); terminating"

I think the issue here may be that the first time I did this, I didn't have wezterm running at all on the remote box.  Now, after the reboot, I started up ~10 local terminals, before I tried doing the "connect" from my laptop.  So there are wezterms running there, but not the mux server.  I saw references in other discussions about running "wezterm-mux-server", but I don't have one of those.  I grabbed the AppImage, which is just the one executable

I can't for the life of me figure out how to get it started again.  The first time went so smoothly...

I'm connecting from MacOS to a Linux machine running Ubuntu 22.04.  My MacOS wezterm config is:

`return {
  ssh_domains = {
    {
      name = "nlaptop",
      remote_address = "nlaptop",
      local_echo_threshold_ms = 400,
      multiplexing = "WezTerm",
      remote_wezterm_path = "/home/sean/bin/wezterm",
      username = "sean",
    }
  },
}`

Thanks for any help you can provide.

**Comment by wez** [comment, marked as answer]
When you say you have 10 wezterms running on the local host, do you mean running on the mac machine or running on the linux system where you want to connect to the multiplexer?

What normally happens in this flow is that the GUI client will try to run `wezterm cli --prefer-mux proxy` on the remote host when you `wezterm connect` to an ssh domain.  That should in turn notice that the mux server isn't running and spawn it.

On the linux machine: the mux server will try to log to file(s) named `/run/user/1000/wezterm/wezterm-mux-server-log-*.txt` or `/run/user/1000/wezterm/wezterm-log-*.txt`; that might have some useful context on why it isn't running.

I would suggest also trying:
* ssh into the linux box
* run `wezterm cli --prefer-mux spawn` to see if it can successfully start a new process via the mux server.

**Reply by linsomniac** [reply]
The 10 wezterms are running on the Linux machine.  IOW: A week ago I installed wezterm on both machines and did "wezterm connect nlaptop" (the Linux machine is a laptop, to clarify).  Then I did a reboot, started 10 wezterms on the Linux machine, and then tried a "wezterm connect nlaptop" and it failed with the "No such file" error trying to open the sock.

I meant to add the full output at the end of my question above, but I forgot I'm afraid.

Running "wezterm cli --prefer-mux spawn" on the Linux machine seems to have worked:

    06:48:21.088  WARN   wezterm_client::client > While connecting to Socket("/run/user/1000/wezterm/sock"): connecting to /run/user/1000/wezterm/sock.  Will try spawning the server.
    06:48:21.088  WARN   wezterm_client::client > Running: "/tmp/.mount_wezterhOhUeC/usr/bin/wezterm-mux-server" "--daemonize"

And then "wezterm connect nlaptop" from the Mac works, thanks!

Should I file this as an issue?  I was also thinking of making a PR for the Multiplexing portion of the documentation mentioning "wezterm cli --prefer-mux proxy" for manually starting the multiplexer, do you think that would be useful?

Thanks!  Loving wezterm!

**Reply by wez** [reply]
Please file an issue so we can figure out how to recreate the scenario that didn't work!
For ssh domains it is not expected that you would need to manually start the mux server, so it sounds like there is case that isn't being handled that we need to run down.
